src/lib/sql/sql_pipeline.hpp

```diff
@@ -48,29 +39,18 @@ class SQLPipeline : public Noncopyable {
               const std::shared_ptr<PreparedStatementCache>& prepared_statements,
               const CleanupTemporaries cleanup_temporaries);
 
-  // Returns the SQL string for each statement.
+  // Vectors containing one element for each SQL statement
   const std::vector<std::string>& get_sql_strings();
-
-  // Returns the parsed SQL string for each statement.
   const std::vector<std::shared_ptr<hsql::SQLParserResult>>& get_parsed_sql_statements();
-
-  // Returns the unoptimized LQP root for each statement.
   const std::vector<std::shared_ptr<AbstractLQPNode>>& get_unoptimized_logical_plans();
-
-  // Returns the optimized LQP root for each statement.
```

Comment:
So, given the "Vectors containing one element for each SQL statement" above, you would still have the individual comments here for each function here??

Reply:
Yes. Frankly, I would leave all of these comments. While I am not a fan of superfluous inline comments (`// increase loop variable by one`), I see a great value in textual comments on a method level.

Reply:
Added them again. But look at this

![image](https://user-images.githubusercontent.com/2679018/48259171-74d07780-e417-11e8-9660-3542959b3a6f.png)

this very simple getter-doc in the SQLPipelineStatement was incorrect and made the reader think there are potentially multiple LQPs in a statement. If the meaning of a comment can be inferred from the code, I would omit the comment.

Reply:
There are probably dozens and dozens of documenting comments in Hyrise, each with good potential of confusing a reader. If we remove the comments that doc somewhat obvious stuff, then there is less potential for confusion ;)